File: vercel-deploy/api/health.js
// Vercel Serverless Function for Health Check
export default async function handler(req, res) {
    // 设置CORS头
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    if (req.method === 'OPTIONS') {
        return res.status(200).end();
    }

    if (req.method !== 'GET') {
        return res.status(405).json({ error: 'Method not allowed' });
    }

    try {
        const healthData = {
            status: 'ok',
            message: '飞书分享代理服务器运行正常',
            timestamp: new Date().toISOString(),
            uptime: process.uptime(),
            version: '1.0.0',
            endpoints: {
                oauth_start: '/api/oauth/start',
                oauth_callback: '/api/oauth/callback',
                oauth_status: '/api/oauth/status/{state}',
                proxy: '/api/proxy'
            },
            environment: {
                node_version: process.version,
                platform: process.platform,
                memory_usage: process.memoryUsage()
            }
        };

        return res.status(200).json(healthData);

    } catch (error) {
        console.error('Health check error:', error);
        return res.status(500).json({
            status: 'error',
            message: '服务器内部错误',
            error: error.message,
            timestamp: new Date().toISOString()
        });
    }
}

File: vercel-deploy/api/health.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
健康检查端点 - Vercel Serverless Function
"""

import json

def handler(request):
    """健康检查"""
    return {
        'statusCode': 200,
        'headers': {'Access-Control-Allow-Origin': '*'},
        'body': json.dumps({
            'status': 'ok', 
            'message': '飞书分享代理服务器运行正常 (Vercel Python)',
            'version': '2.0.0'
        })
    }

# Vercel entry point
def main(request):
    return handler(request)
